moto/databrew/responses.py

```diff
@@ -50,12 +50,98 @@ def list_recipes(self):
             }
         )
 
+    def put_recipe_response(self, recipe_name):
+        recipe_description = self.parameters.get("Description")
+        recipe_steps = self.parameters.get("Steps")
+        tags = self.parameters.get("Tags")
+
+        recipe = self.databrew_backend.update_recipe(
+            recipe_name, recipe_description, recipe_steps, tags
+        )
+        return 200, {}, json.dumps(recipe.as_dict())
+
+    def get_recipe_response(self, recipe_name):
+        recipe = self.databrew_backend.get_recipe(recipe_name)
+        return 201, {}, json.dumps(recipe.as_dict())
+
     @amzn_request_id
-    def describe_recipe_response(self, request, full_url, headers):
+    def recipe_response(self, request, full_url, headers):
         self.setup_class(request, full_url, headers)
         parsed_url = urlparse(full_url)
 
         recipe_name = parsed_url.path.rstrip("/").rsplit("/", 1)[1]
 
-        recipe = self.databrew_backend.get_recipe(recipe_name)
-        return json.dumps(recipe.as_dict())
+        if request.method == "PUT":
+            return self.put_recipe_response(recipe_name)
+        elif request.method == "GET":
+            return self.get_recipe_response(recipe_name)
+
+    @amzn_request_id
+    def create_ruleset(self):
+        ruleset_description = self.parameters.get("Description")
+        ruleset_rules = self.parameters.get("Rules")
+        ruleset_name = self.parameters.get("Name")
+        ruleset_target_arn = self.parameters.get("TargetArn")
+        tags = self.parameters.get("Tags")
+
+        return json.dumps(
+            self.databrew_backend.create_ruleset(
+                ruleset_name,
+                ruleset_description,
+                ruleset_rules,
+                ruleset_target_arn,
+                tags,
+            ).as_dict()
+        )
+
+    def put_ruleset_response(self, ruleset_name):
+        ruleset_description = self.parameters.get("Description")
+        ruleset_rules = self.parameters.get("Rules")
+        tags = self.parameters.get("Tags")
+
+        ruleset = self.databrew_backend.update_ruleset(
+            ruleset_name, ruleset_description, ruleset_rules, tags
+        )
+        return 200, {}, json.dumps(ruleset.as_dict())
+
+    def get_ruleset_response(self, ruleset_name):
+        ruleset = self.databrew_backend.get_ruleset(ruleset_name)
+        return 201, {}, json.dumps(ruleset.as_dict())
+
+    def delete_ruleset_response(self, ruleset_name):
+        self.databrew_backend.delete_ruleset(ruleset_name)
+        return 204, {}, ""
+
+    @amzn_request_id
+    def ruleset_response(self, request, full_url, headers):
+        self.setup_class(request, full_url, headers)
+        parsed_url = urlparse(full_url)
+
+        ruleset_name = parsed_url.path.rstrip("/").rsplit("/", 1)[1]
```

Comment:
```suggestion
        ruleset_name = parsed_url.path.split("/")[-1]
```
Assuming that the ruleset_name is always last in the URL, this would be easier to read

Reply:
Agreeed, though I copied that code from else where in moto. Though maybe it's use case is different. So will use as suggested.